### ISSUE TITLE
Is validation data necessary?

### ISSUE THREAD

**Issue by 592319702** [ml-explore/mlx-examples#527]
If I use my own data to fine-tune with lora, is validation data necessary? Because validation data is used to train hyperparameters, our framework have already pre-defined the hyperparameters. So I think maybe validation data is unnecessary?

**Comment by awni** [comment]
It's quite useful to log validation loss for early stopping as well. It's easy to overfit, especially with large models and small datasets and the validation set is the best way to determine...

Feel free to open a discussion if you'd like more input on this.

**Comment by 592319702** [comment]
> Member

I notice the Val loss will be printed every 200 iterations, how can I make it print more frequent?

**Comment by awni** [comment]
`--steps-per-eval=100` for example

**Comment by awni** [comment]
`python -m mlx_lm.lora --help` for all the options
